src/common/HexHighlighter.cpp

```diff
@@ -27,51 +27,53 @@ HexHighlighter::HexHighlighter(QTextDocument *parent)
                     << "\\b75\\b" << "\\b76\\b" << "\\b77\\b" << "\\b78\\b" << "\\b79\\b" << "\\b7a\\b" << "\\b7b\\b"
                     << "\\b7c\\b" << "\\b7d\\b" << "\\b7e\\b" << "\\b7f\\b";
     for (const QString &pattern : keywordPatterns) {
-        rule.pattern = QRegExp(pattern);
-        rule.pattern.setCaseSensitivity(Qt::CaseInsensitive);
+        rule.pattern = pattern;
+        rule.options = QRegularExpression::CaseInsensitiveOption;
         rule.format = keywordFormat;
         highlightingRules.append(rule);
     }
 
     singleLineCommentFormat.setFontWeight(QFont::Bold);
     singleLineCommentFormat.setForeground(Qt::darkGreen);
-    rule.pattern = QRegExp(";[^\n]*");
+    rule.pattern = ";[^\n]*";
     rule.format = singleLineCommentFormat;
     highlightingRules.append(rule);
 
-    commentStartExpression = QRegExp("/\\*");
-    commentEndExpression = QRegExp("\\*/");
+    commentStartRegularExpression = "/\\*";
+    commentEndRegularExpression = "\\*/";
```

Comment:
This can affect performance too

Reply:
```suggestion
    commentStartRegularExpression = QStringLiteral("/\\*");
    commentEndRegularExpression = QStringLiteral("\\*/");
```
Do you have this in mind?

Reply:
well, with 8979dddf1c0b1c3fd770ad2c272f4fced26654a8 that is obsolete